pylearn2/models/multisubject.py

```diff
@@ -555,21 +575,25 @@ def cost_from_X(self, data):
         ----------
         data : WRITEME
         """
-        self.cost_from_X_data_specs()[0].validate(data)
-        X, Y = data
-        Y_hat = self.fprop(X)
-        return self.cost(Y, Y_hat)
+
+        for mlp, ds in safe_izip(model.mlps, data): 
+            self.cost_from_X_data_specs()[0].validate(data)
+            X, Y = data
+            Y_hat = self.fprop(X)
+            return self.cost(Y, Y_hat)
 
     def cost_from_X_data_specs(self):
         """
         Returns the data specs needed by cost_from_X.
 
         This is useful if cost_from_X is used in a MethodCost.
         """
-        space = CompositeSpace((self.get_input_space(),
-                                self.get_target_space()))
-        source = (self.get_input_source(), self.get_target_source())
-        return (space, source)
+        
+        for mlp, ds in safe_izip(model.mlps, data): 
+            space = CompositeSpace((self.get_input_space(),
+                                    self.get_target_space()))
+            source = (self.get_input_source(), self.get_target_source())
+            return (space, source)
```

Comment:
Here, I think the MultisubjectMLP will have a CompositeSpace which is made up of the CompositeSpaces of the individual MLPs.